pkg/sql/executor.go

```diff
@@ -1148,6 +1150,9 @@ func (e *Executor) execStmtsInCurrentTxn(
 		// always guaranteed to execute regardless of the current transaction state.
 		if _, ok := stmt.AST.(*parser.ShowTransactionStatus); ok {
 			res, err = runShowTransactionState(session)
+
+			session.removeActiveQuery(queryID)
+			e.cfg.QueryRegistry.deregister(queryID)
```

Comment:
What is the point of dragging these both statements everywhere? At least combine them into one function? Or just cancel everything inside `FinishPlan`